Sources/BraveWallet/Settings/IPFSCustomGatewayView.swift

```diff
@@ -0,0 +1,127 @@
+    self.ipfsAPI = ipfsAPI
+  }
+  
+  var body: some View {
+    List {
+      Section(footer: Text(Strings.Wallet.nftGatewayLongDescription)) {
+        TextField(ipfsAPI.nftIpfsGateway?.absoluteString ?? "", text: $url)
+          .font(.body)
+          .keyboardType(.URL)
+          .autocapitalization(.none)
+          .disableAutocorrection(true)
+          .foregroundColor(Color(.braveLabel))
+          .onChange(of: url) { newValue in
+            if newValue.count < 8 && "https://".hasPrefix(newValue) {
+              self.url = "https://"
+              setButtonStatus = .disabled
+            } else {
+              // The withSecureUrlScheme is used in order to force user to use secure url scheme
+              // Instead of checking paste-board with every character entry, the textView text is analyzed
+              // and according to what prefix copied or entered, text is altered to start with https://
+              // this logic block repeating https:// and http:// schemes
+              let textEntered = newValue.withSecureUrlScheme
+              self.url = textEntered
+              
+              if let enteredURL = URL(string: textEntered), enteredURL != ipfsAPI.nftIpfsGateway {
+                setButtonStatus = .enabled
+              } else {
+                setButtonStatus = .disabled
+              }
+            }
+          }
+          .listRowBackground(Color(.secondaryBraveGroupedBackground))
+      }
+    }
+    .listStyle(InsetGroupedListStyle())
+    .listBackgroundColor(Color(UIColor.braveGroupedBackground))
+    .navigationBarTitleDisplayMode(.inline)
+    .navigationTitle(Strings.Wallet.nftGatewayTitle)
+    .navigationBarItems(
+      // Have to use this instead of toolbar placement to have a custom button style
+      trailing: Button(action: {
+        validateURLAndSet()
+      }) {
+        if setButtonStatus == .loading {
+          ProgressView()
+        } else {
+          Text(Strings.Wallet.setGatewayButtonTitle)
+        }
+      }
+        .buttonStyle(BraveFilledButtonStyle(size: .small))
+        .disabled(setButtonStatus != .enabled)
+    )
+    .alert(isPresented: $presentWrongGatewayAlert) {
+      Alert(
+        title: Text(Strings.Wallet.wrongGatewayAlertTitle),
+        message: Text(Strings.Wallet.wrongGatewayAlertDescription),
+        dismissButton: .cancel(Text(Strings.OKString))
+      )
+    }
+    .onAppear {
+      if let url = ipfsAPI.nftIpfsGateway?.absoluteString {
+        self.url = url
+      }
+    }
+  }
+  
+  private func validateURLAndSet() {
+    guard let enteredURL = URL(string: url), enteredURL.isSecureWebPage() else {
+      presentWrongGatewayAlert = true
+      return
+    }
+    var testURL = enteredURL
+    testURL.append(pathComponents: IPFSCustomGatewayView.ipfsTestPath, IPFSCustomGatewayView.ipfsTestCID)
+    
+    setButtonStatus = .loading
+    resignFirstResponder()
+    
+    URLSession.shared.dataTask(with: testURL) { data, response, error in
```

Comment:
Can we use async/await apis here?

Reply:
fixed in [fa59bc5](https://github.com/brave/brave-ios/pull/6880/commits/fa59bc5e8fdf66b6faffc6228672cfa2b695c87a)